azurerm/resource_arm_hdinsight.go

```diff
@@ -0,0 +1,888 @@
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"isdefault": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						"container": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"key": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+				Set: resourceArmHDInsightstorageaccountHash,
+			},
+		},
+	}
+}
+
+func resourceArmHDInsightCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	hdiClusterClient := client.hdiClusterClient
+	log.Printf("[INFO] preparing arguments for AzureRM HDI creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	location := d.Get("location").(string)
+	clusterVersion := d.Get("cluster_version").(string)
+	tags := d.Get("tags").(map[string]interface{})
+	RolesProfile, err := expandAzureHDInsightRoleProfile(d)
+	storageProfile := expandAzureHDInsightStorageProfile(d)
+
+	tier := hdinsight.Premium
+	switch d.Get("tier").(string) {
+	case "Prenium":
+		tier = hdinsight.Premium
+		break
+	case "Standard":
+		tier = hdinsight.Standard
+		break
+	}
+
+	clusterDefinition, err := expandAzureHDInsightClusterDefinition(d)
+
+	clusterCreateProperties := hdinsight.ClusterCreateProperties{
+		ClusterVersion:    &clusterVersion,
+		OsType:            hdinsight.Linux,
+		Tier:              tier,
+		ClusterDefinition: clusterDefinition,
+		ComputeProfile: &hdinsight.ComputeProfile{
+			Roles: RolesProfile,
+		},
+	}
+
+	securityProfile, err := expandAzureHDInsightSecurityProfile(d)
+	if securityProfile != nil {
+		print(securityProfile)
+		clusterCreateProperties.SecurityProfile = securityProfile
+	}
+
+	if storageProfile != nil {
+		clusterCreateProperties.StorageProfile = storageProfile
+	}
+	//	fmt.Printf("clusterCreateProperties :%v", clusterCreateProperties)
+	parameters := hdinsight.ClusterCreateParametersExtended{
+		Location:   &location,
+		Tags:       expandTags(tags),
+		Properties: &clusterCreateProperties,
+	}
+
+	fmt.Printf("Parameters: %+v\n", clusterCreateProperties)
+	//ttlInSeconds := "60"
+	ctx := meta.(*ArmClient).StopContext
+
+	createFuture, err := hdiClusterClient.Create(ctx, resGroup, name, parameters)
+
+	//fmt.Printf("createFuture: %+v\n", createFuture.Future.Response)
+	if err != nil {
+		return err
+	}
+
+	err = createFuture.WaitForCompletion(ctx, hdiClusterClient.Client)
+	if err != nil {
+		return err
+	}
+
+	read, err := hdiClusterClient.Get(ctx, resGroup, name)
+	if err != nil {
+		return err
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read HDI %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmHDInsightRead(d, meta)
+}
+
+/*
+func resourceArmHDInsightUpdate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
it looks like it's possible to update the size/count of the machines; so can we add this back in at some point?